testing/btest/Baseline.cpp/bifs.to_addr/error

```diff
@@ -1,2 +1,2 @@
 ### BTest baseline data generated by btest-diff. Do not edit. Use "btest -U/-u" to update. Requires BTest >= 0.63.
-error in <no location>: failed converting string to IP address (not an IP)
+error in <...>/to_addr.zeek (C++), line 12: failed converting string to IP address (Users_vern_warehouse_zeek_zeek_master_02Mar23_testing_btest__tmp_bifs_to_addr_to_addr_zeek__global_stmts__zf() and not an IP)
```

Comment:
This looks a bit unusual:
* The original baseline only has "not an IP" in parenthesis, this one has some path encoded form of a function, too (for the other error reports as well). If that's valuable information, maybe it should be somewhere close to the `(C++)` part?
* There seems to be user/directory specific output in these baselines (judging by /Users/vern/warehouse). Might need to add a canonifier (or suppress that variable part during generation) such that running the tests in other directories, in CI, or by other people won't fail.

Reply:
Regarding the first: yeah, this is a side effect of using a slip-shod method of tracking calls in scripts compiled to C++.  Previously, this would have reported `no location` (as shown in the diff), and in some cases actually caused crashes.  Now, it provides the root of the C++ code that eventually led to the problematic call.  Not great, of course, though significantly better than before ... and reflecting a conscious design decision that `-O gen-C++` emphasizes performance over error pinpointing, with the model being that in general you should only compile scripts to C++ after being satisfied with their correctness.

Regarding the second point, thanks for flagging.  I've implemented a canonifier.  The question is now how to integrate it into the BTest framework.  It seems a pity to alter the BTests itself to use it, since in general there's no obvious reason which ones need it versus something else (for example, for `bifs.to_addr` above).  I can simply extend `diff-remove-abspath` to include this as part of its removal, with a comment to that effect - does that sound like the best way to go?  (If we had `$SCRIPTS` search paths then we could add a new directory for auxiliary scripts, e.g. `testing/CPP-scripts`; but without search paths that doesn't work since it won't have the other stuff in it that's needed by other BTests.)

BTW, the addition looks like:
```
 $sed 's#/+#/#g' |
+    $sed 's#[^( ]*testing_btest#testing_btest#' |
     $sed 's#([a-zA-Z]:)?/([^	 :/]{1,}/){1,}([^	 :/]{1,})#<...>/\3#g'
```
which IMHO is modest enough that adding that to the current `scripts/diff-remove-abspath` seems ok.

Reply:
No good intuition :-(

If canonifiers were used solely from the PATH, things would be easier, but that's not the layout, so, hmm.

Could possibly set SCRIPTS to a directory specifically for the CPP environment, but it would require replicating the old dir with a bunch of symlinks or platform specific overlay tricks to make this work quite right. Neither seems attractive.

The other thought would be to normalize the paths to the `cwd` or some other anchor within Zeek. I haven't really looked where it's coming from or if that's sensible, but with std::filesystem (as zeek::fileystem) being available there should be helpers so it's not too painful.

But, not opposed to the canonifier either. Suggest to add a `<...>` to indicate the replacement better for when it triggers unexpectedly?  Or maybe `<--->` or to differentiate from the existing one.

I'm resolving this assuming canonifier. If someone else has stronger opinions can reopen.

Reply:
Okay I've pushed a change to the canonifier.  Good thought re a marker, I went with `<___>` because what's removed is generally a bunch of tokens separated by `_`'s.